main.go

```diff
@@ -81,26 +98,26 @@ func main() {
 	// for parallel execution. Setting the maximum number of CPUs to the available
 	// core count ensures the application can make full use of all available hardware.
 	// If multithreading is not enabled, server will run on a single core.
-	var numCores int
+	var numShards int
 	if config.EnableMultiThreading {
-		numCores = runtime.NumCPU()
+		numShards = runtime.NumCPU()
 		if config.NumShards > 0 {
-			numCores = config.NumShards
+			numShards = config.NumShards
 		}
-		logr.Debug("The DiceDB server has started in multi-threaded mode.", slog.Int("number of cores", numCores))
+		slog.Info("running with", slog.String("mode", "multi-threaded"), slog.Int("num-shards", numShards))
 	} else {
-		numCores = 1
-		logr.Debug("The DiceDB server has started in single-threaded mode.")
+		numShards = 1
+		slog.Info("running with", slog.String("mode", "single-threaded"))
 	}
 
-	// The runtime.GOMAXPROCS(numCores) call limits the number of operating system
+	// The runtime.GOMAXPROCS(numShards) call limits the number of operating system
 	// threads that can execute Go code simultaneously to the number of CPU cores.
 	// This enables Go to run more efficiently, maximizing CPU utilization and
 	// improving concurrency performance across multiple goroutines.
-	runtime.GOMAXPROCS(numCores)
+	runtime.GOMAXPROCS(runtime.NumCPU())
```

Comment:
Shall this be `numShards`?

Reply:
The max proc should be max cores while shards can be fewer than cores. this just limits the number of processes on which it will be multiplxed